### ISSUE TITLE
Error message: cannot assign requested address

### ISSUE THREAD

**Issue by cc13com** [Hentra/dyndns-netcup-go#16]
Hello,

I'm runing the Docker image on my RPi4 and getting the following message after starting the container:

```
2023/08/07 11:40:43 configure DNS records
2023/08/07 11:40:44 Get "https://api6.ipify.org?format=text": dial tcp [2607:xxxxxxxx]:443: connect: cannot assign requested address
```

Any idea to solve this?

**Comment by martin-georgi** [comment]
Have you tried ` --network host `?  This fixed it on my pi4.

**Comment by axodentally** [comment]
This is probably because docker networks are ipv4 only by default.
See [here](https://docs.docker.com/engine/daemon/ipv6/) for the official documentation or [here](https://ounapuu.ee/posts/2024/12/20/docker-ipv6/) for a good blog post about it.

TLDR: use a compose file that looks something like this:
```yaml
services:
  dyndns-netcup:
	...
    networks:
      - dyndns-netcup-network

networks:
  dyndns-netcup-network:
    enable_ipv6: true
```

That should work without the need for using `--network host`, which is considered bad practice almost all of the time in docker networking.
